src/simulations/set_generator.py

```diff
@@ -331,29 +332,33 @@ def __iter__(self):
       yield set_ids_list[i]
     return self
 
+
 class _SequentiallyCorrelatedAllPreviousSetGenerator(SetGeneratorBase):
   """Generator for sequentailly correlated sets.
 
   Each set has some overlap with the union of previously generated campaigns.
   """
 
-  def __init__(self, universe_size, shared_prop, set_size_list, random_state):
-    self.universe_size = universe_size
+  def __init__(self, shared_prop, set_size_list, random_state):
     self.random_state = random_state
     self.union_ids = np.array([], dtype=int)
     self.set_size_list = set_size_list
-    self.overlap_size_list = ([0]
-                              + [int(set_size * shared_prop)
-                                 for set_size in self.set_size_list[:-1]]
-                              + [0])
-    # self.overlap_size_list[i] is the overlap of set.set_size_list[i]
-    # with the previous union, so in particular, self.overlap_size_list[0] = 0
-    total_ids_size = int(sum(self.set_size_list) - sum(self.overlap_size_list))
-    self.ids_pool = choice_fast(universe_size, total_ids_size, self.random_state)
+    self.num_sets = len(set_size_list)
+    self.overlap_size_list = [0]
+    total_ids_size = set_size_list[0]
+    for i in range(self.num_sets - 1):
```

Comment:
Can you add comments about what you're doing here. Basically just transfer the description from the CL. I think it's not straightforward why this code exists.

Reply:
Done.